include/Webservices/CustomerPortalWS.php

```diff
@@ -864,6 +864,85 @@ function getReferenceAutocomplete($term, $filter, $searchinmodules, $limit, $use
 	return $respuesta;
 }
 
+/**
+ * @param String $term: search term
+ * @param Array $returnfields: array of fields to return as result, maybe for the future
+ * @param Number $limit: maximum number of values to return
+ * @return Array values found
+ */
+function getProductServiceAutocomplete($term, $returnfields = array(), $limit = 5) {
+	global $adb, $current_user;
+	require_once 'include/fields/CurrencyField.php';
+	require_once 'include/utils/CommonUtils.php';
+
+	$r = $adb->query("SELECT * FROM 
+		(SELECT 
+		    vtiger_products.productname AS name, 
+		    vtiger_products.divisible AS divisible, 
+		    'Products' AS type, 
+		    vtiger_products.vendor_part_no AS ven_no, 
+		    vtiger_products.cost_price AS cost_price, 
+		    vtiger_products.mfr_part_no AS mfr_no, 
+		    vtiger_products.qtyinstock AS qtyinstock, 
+		    vtiger_products.discontinued AS discontinued, 
+		    vtiger_crmentity.description AS description, 
+		    vtiger_crmentity.deleted AS deleted, 
+		    vtiger_crmentity.crmid AS id, 
+		    vtiger_products.unit_price AS unit_price 
+		    FROM vtiger_products 
+		    INNER JOIN vtiger_crmentity ON vtiger_products.productid = vtiger_crmentity.crmid 
+		    WHERE vtiger_products.productname LIKE '%{$term}%' 
+		    OR vtiger_products.mfr_part_no LIKE '%{$term}%' 
+		    OR vtiger_products.vendor_part_no LIKE '%{$term}%' 
+		    UNION
+		SELECT
+		    vtiger_service.servicename AS name, 
+		    vtiger_service.divisible AS divisible, 
+		    'Services' AS type,
+		    '' AS ven_no, 
+		    '' AS mfr_no,
+		    0 AS qtyinstock,
+		    '' AS cost_price,
+		    vtiger_service.discontinued AS discontinued, 
+		    vtiger_crmentity.description AS description, 
+		    vtiger_crmentity.deleted AS deleted, 
+		    vtiger_crmentity.crmid AS id, 
+		    vtiger_service.unit_price AS unit_price 
+		    FROM vtiger_service 
+		    INNER JOIN vtiger_crmentity ON vtiger_service.serviceid = vtiger_crmentity.crmid 
+		    WHERE vtiger_service.servicename LIKE '%{$term}%'
+		) AS prod_ser 
+		    WHERE prod_ser.deleted = 0 AND prod_ser.discontinued = 1 LIMIT $limit");
+	$ret = array();
+
+	while ($prodser = $adb->fetch_array($r)) {
+		$ret_prodser = array();
+		$ret_prodser['meta'] = array(
+			'image' => '',
+			'name' => $prodser['name'],
+			'divisible' => $prodser['divisible'],
+			'comments' => $prodser['description'],
+			'ven_no' => $prodser['ven_no'],
+			'mfr_no' => $prodser['mfr_no'],
+			'type' => $prodser['type'],
+			'id' => $prodser['id'],
+		);
+		$ret_prodser['pricing'] = array(
+			'unit_price' => number_format($prodser['unit_price'], 2, '.', ''),
```

Comment:
These are the lines that trigger the warning, probably the unit_price/cost are empty or something??